addons/godot-xr-tools/objects/snap_zone.gd

```diff
@@ -281,6 +281,14 @@ func _update_snap_mode() -> void:
 
 # Called when a target in our grab area is dropped
 func _on_target_dropped(target: Spatial) -> void:
+	# The target may have been dropped out of a different snap-zone to be picked
+	# up by the player. Defer our pickup until idle, and only pick up the object
+	# if it's still available.
+	call_deferred("_try_grab_dropped", target)
```

Comment:
Instead of this, you can connect to a signal and mark it as deferred and it will do this automatically.

Reply:
Updated to defer in the connect method.